### PR TITLE
Rework docs: separate rarely used features

### PR DESCRIPTION
**Pull request by e2**
Re-created PR on `rework-cukes-integration` branch from here:

https://github.com/rspec/rspec-core/pull/2262#issuecomment-223833012

### CODE REVIEW

File: features/.nav [hunk 1]
```diff
@@ -44,22 +44,21 @@
...
-  - use_any_framework.feature
 - formatters:
   - text_formatter.feature
   - configurable_colors.feature
   - json_formatter.feature
   - custom_formatter.feature
-- spec_files:
-  - arbitrary_file_suffix.feature
 - core_standalone.feature
+- legacy_support_and_special_project_needs
```
Comment (myronmarston) [inline features/.nav]: So, a few ideas here:

* "Using RSpec with Alternate Tools"
* "Usage with Other Testing Libraries"
* "Usage with Alternate Tools"

Although, all 3 of those ideas would probably not fit `arbitrary_file_suffix` or `alias_example_to` but I think that's OK.  We can find a better place for those two.

Also, I don't think we need `use_rspec_mocks.feature` -- the default behavior is to use rspec-mocks, so why should we take up valuable doc space with this?  We'd need to make sure the examples in that file are covered by specs, of course.

File: features/.nav [hunk 2]
```diff
@@ -44,22 +44,21 @@
...
+  - arbitrary_file_suffix.feature
+  - alias_example_to.feature
+  - use_rspec_mocks.feature
+  - use_flexmock.feature
+  - use_mocha.feature
+  - use_rr.feature
+  - use_any_framework.feature
```
Comment (myronmarston) [inline features/.nav]: Should this be `use_any_mocking_library.feature`?

**Comment by myronmarston** [comment]
It looks like you only re-worked the `.nav` but haven't moved the files themselves around.  It'd be nice to keep them in sync (I think it's confusing if our source-controlled structure doesn't match the published structure).  That said, moving files around is a lot more likely to create merge conflicts than `.nav` changes are -- so it might make sense to not move the files now, and do all the file moves in one final commit for the entire project once everything else is finalized before merging everything to master.

Thoughts?

File: features/.nav (hunk 2 above)
Comment (myronmarston) [inline features/.nav]: Also, it's probably worth updating the `Feature` line of the feature files to match the file name (or vice versa).  Regardless, keeping them in sync would be nice.

File: features/.nav (hunk 1 above)
Comment (e2) [inline features/.nav]: Done. 

I chose "alternate tools and libraries", so it sounds general enough for people to know to look there if needed, while still having a single category.

Without much thinking, I lazily converted the feature file into an integration tests - I'm sure you'll have lots of comments, so fire away.

I moved the file_suffix and alias_example into a separate category: "legacy_features_and_workarounds".

**Comment by e2** [comment]
I think deciding on the structure is going to need more flexibility anyway - so I think I prefer to re-do the commits if needed. There are too many unknowns at this point to worry about merge conflicts. And the commits aren't really too descriptive anyway.

It would probably be better to just regenerate the ".nav" based on feature tags instead - because that's where merge conflicts will likely be.

I don't mind moving files. Once we agree on a batch of work to merge, it will be easier to know what merge conflicts to expect and how to avoid them. I don't mind interactive rebasing and with new folders created I don't think there will be too many on the filename/dirname side.

File: features/.nav [hunk 3]
```diff
@@ -44,22 +44,21 @@
...
 - core_standalone.feature
+- using_alternate_tools_and_libraries:
+  - use_flexmock.feature
+  - use_mocha.feature
+  - use_rr.feature
+  - use_any_framework.feature
+  - configure_expectation_framework.feature
+- legacy_features_and_workarounds:
+  - arbitrary_file_suffix.feature
+  - alias_example_to.feature
```
Comment (JonRowe) [inline features/.nav]: I'm not sure either of these are considered legacy or workarounds.

Comment (e2) [inline features/.nav]: Not sure what else to call them. They're not really day to day work type features. I'd call it "weird stuff" but that's inappropriate...

Comment (JonRowe) [inline features/.nav]: They're configuration, put them back there.

Comment (myronmarston) [inline features/.nav]: Personally, I think they are TBD.  So maybe just create a category called "tbd" and put them in there for now?  That can be for everything we haven't found a place for yet.

Comment (myronmarston) [inline features/.nav]: The nice thing about `tbd` as a category is it ensures we don't forget to address those, kinda like variables you temporarily name `fixme` or whatever.

**Comment by e2** [comment]
Yes, I like that!

**Comment by myronmarston** [comment]
One other thing....@e2, I find it extremely useful to be able to push to a relish staging area when I work on the cukes so I can see how they render, detect mistakes early, etc.  I'd encourage you to do the same.  (For example, if the file names of the feature files don't match what's in `.nav`, I think relish will give you errors).

I wrote a task for this that I use:

https://github.com/rspec/rspec-core/blob/bc1482605763cc16efa55c98b8da64a89c8ff5f9/Rakefile#L71-L73

...but you have to have access to the rspec-staging user on relish and I don't think there's an easy way to give you access (I asked once).  So maybe it would work best to make your own account and you can change the rake task to support the staging user being overridable via an ENV var or something so you can push to it?

That way, when you open a PR or make changes you can provide us with a link to go look at what the results render like on relish.

Thoughts?

**Comment by e2** [comment]
> I'd encourage you to do the same.

Yes! I wanted to ask about that. Being able to push myself would help avoid lots of back-and-forth rework too.

**Comment by myronmarston** [comment]
Hey @e2 do you plan to get back to this?  Given the lack of activity for the last month I'm going to close this but will gladly reopen if you decide to pick this back up.

**Comment by e2** [comment]
I'm crushed with work after my vacation. No time for this now, but will reopen when I update the PR.

**Pull request closed without merging**